File: my-solutions/0283-move-zeroes/solution.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int zero_count = 0;

        for (int i = 0; i < nums.size(); ++i) {
            if (nums[i] == 0) {
                nums.erase(nums.begin() + i);
                --i;
                ++zero_count;
            }
        }

        for (int i = 0; i < zero_count; ++i) {
            nums.push_back(0);
        }
    }
};

File: my-solutions/0283-move-zeroes/solution.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        slotPointer = 0        
        for pointer in range(len(nums)):
            if nums[pointer] != 0:
                nums[pointer], nums[slotPointer] = nums[slotPointer], nums[pointer]
                slotPointer += 1
        
